torch/testing/_internal/mps_utils.py

```diff
@@ -0,0 +1,988 @@
+    "div": MPSSkipInfo(
+        dtypes=COMPLEX_DTYPES,
+        upper=15.0,
+        variant=" ",
+    ),
+    "gather": MPSSkipInfo(dtypes=COMPLEX_DTYPES),
+    "index_add": MPSSkipInfo(dtypes=COMPLEX_DTYPES),
+    "index_fill": MPSSkipInfo(dtypes=COMPLEX_DTYPES),
+    "index_put": MPSSkipInfo(dtypes=COMPLEX_DTYPES),
+    "istft": MPSSkipInfo(dtypes=COMPLEX_DTYPES),
+    "lerp": MPSSkipInfo(dtypes=COMPLEX_DTYPES),
+    "linalg.cholesky": MPSSkipInfo(dtypes=COMPLEX_DTYPES),
+    "linalg.cholesky_ex": MPSSkipInfo(dtypes=COMPLEX_DTYPES),
+    "linalg.cross": MPSSkipInfo(dtypes=COMPLEX_DTYPES),
+    "linalg.det": MPSSkipInfo(dtypes=COMPLEX_DTYPES),
+    "linalg.inv": MPSSkipInfo(dtypes=COMPLEX_DTYPES),
+    "linalg.inv_ex": MPSSkipInfo(dtypes=COMPLEX_DTYPES),
+    "linalg.lu_factor": MPSSkipInfo(dtypes=COMPLEX_DTYPES),
+    "linalg.lu_factor_ex": MPSSkipInfo(dtypes=COMPLEX_DTYPES),
+    "linalg.matrix_norm": MPSSkipInfo(dtypes=COMPLEX_DTYPES),
+    "linalg.matrix_power": MPSSkipInfo(dtypes=COMPLEX_DTYPES),
+    "linalg.norm": MPSSkipInfo(dtypes=COMPLEX_DTYPES),
+    "linalg.solve": MPSSkipInfo(dtypes=COMPLEX_DTYPES),
+    "linalg.solve_ex": MPSSkipInfo(dtypes=COMPLEX_DTYPES),
+    "linalg.solve_triangular": MPSSkipInfo(dtypes=COMPLEX_DTYPES),
+    "linalg.tensorinv": MPSSkipInfo(dtypes=COMPLEX_DTYPES),
+    "linalg.tensorsolve": MPSSkipInfo(dtypes=COMPLEX_DTYPES),
+    "linalg.vander": MPSSkipInfo(dtypes=COMPLEX_DTYPES),
+    "linalg.vector_norm": MPSSkipInfo(dtypes=COMPLEX_DTYPES),
+    "logaddexp": MPSSkipInfo(dtypes=COMPLEX_DTYPES),
+    "lu": MPSSkipInfo(dtypes=[torch.complex64]),
+    "lu_unpack": MPSSkipInfo(dtypes=[torch.complex64]),
+    "masked.cumprod": MPSSkipInfo(dtypes=COMPLEX_DTYPES),
+    "masked.cumsum": MPSSkipInfo(dtypes=COMPLEX_DTYPES),
+    "masked.normalize": MPSSkipInfo(dtypes=COMPLEX_DTYPES),
+    "nn.functional.conv3d": MPSSkipInfo(dtypes=COMPLEX_DTYPES),
+    "nn.functional.l1_loss": MPSSkipInfo(dtypes=COMPLEX_DTYPES),
+    "nn.functional.linear": MPSSkipInfo(dtypes=COMPLEX_DTYPES),
+    "nn.functional.normalize": MPSSkipInfo(dtypes=COMPLEX_DTYPES),
+    "nn.functional.padconstant": MPSSkipInfo(
+        dtypes=COMPLEX_DTYPES,
+        upper=15.0,
+    ),
+    "nn.functional.padreflect": MPSSkipInfo(
+        dtypes=COMPLEX_DTYPES,
+        upper=15.0,
+    ),
+    "nn.functional.padreplicate": MPSSkipInfo(
+        dtypes=COMPLEX_DTYPES,
+        upper=15.0,
+    ),
+    "nn.functional.pairwise_distance": MPSSkipInfo(dtypes=COMPLEX_DTYPES),
+    "nn.functional.silu": MPSSkipInfo(dtypes=COMPLEX_DTYPES),
+    "nn.functional.triplet_margin_loss": MPSSkipInfo(dtypes=COMPLEX_DTYPES),
+    "nn.functional.triplet_margin_with_distance_loss": MPSSkipInfo(
+        dtypes=COMPLEX_DTYPES
+    ),
+    "norm": MPSSkipInfo(dtypes=COMPLEX_DTYPES),
+    "normal": MPSSkipInfo(dtypes=COMPLEX_DTYPES, variant="in_place"),
+    "pow": MPSSkipInfo(dtypes=COMPLEX_DTYPES),
+    "rand_like": MPSSkipInfo(dtypes=COMPLEX_DTYPES),
```

Comment:
But it works, isn't it?
```
>>> torch.rand_like(torch.empty(10, device='mps'), dtype=torch.cfloat)
tensor([0.0038+0.9996j, 0.0703+0.2922j, 0.2603+0.8467j, 0.5551+0.2856j,
        0.6743+0.5194j, 0.0124+0.0555j, 0.9165+0.9206j, 0.7336+0.3115j,
        0.3462+0.6868j, 0.1300+0.2724j], device='mps:0')
```

Reply:
Why you marked it resolved? This decorator is missing in current `test_mps.py`, so why do you need to add it here?